File: server/service/src/main/java/by/iba/dto/resp/advertisement/AdvertisementResp.java
package by.iba.dto.resp.advertisement;

import by.iba.common.dto.core.FullAbstractResp;
import by.iba.dto.resp.user.UserResp;
import by.iba.entity.user.AdvertisementStatus;
import by.iba.entity.user.AdvertisementType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@NoArgsConstructor
@Getter
@Setter
public class AdvertisementResp extends FullAbstractResp {

    private String title;

    private String body;

    private String picture;

    private AdvertisementType type;

    private AdvertisementStatus status;

    private Set<AdvertisementRatingResp> ratingSet = new HashSet<>();

    private Double ratingSum;

    private UserResp user;
}

File: client/app/src/main/java/by/chekun/repository/database/entity/advertisement/view/AdvertisementResp.kt
package by.chekun.repository.database.entity.advertisement.view

import by.chekun.repository.database.entity.AbstractDto


class AdvertisementResp() : AbstractDto() {

    var title: String? = null

    var body: String? = null

    var picture: String? = null

    var type: AdvertisementType? = null

    var status: AdvertisementStatus? = null

    fun getStringType():String{
        return type.toString()
    }

    fun getStringStatus():String{
        return status.toString()
    }

    fun getId():String{
        return id.toString()
    }

    fun getStringBody():String{
        return body.toString()
    }

    fun getStringTitle():String{
        return title.toString()
    }


}
